app/controllers/api/v2/user_notification_preferences_controller.rb

```diff
@@ -0,0 +1,31 @@
+module API
+  module V2
+    class UserNotificationPreferencesController < API::V2::ApplicationController
+      deserializable_resource :user_notification_preferences,
+                              class: API::V2::DeserializableUserNotificationPreferences
+
+      def show
+        authorize user_notification_preferences = UserNotificationPreferences.new(user_id: params[:id])
+
+        render jsonapi: user_notification_preferences
+      end
+
+      def update
```

Comment:
I agree with the point in the PR description about trying to remove complexity from the front end. This version does make things simpler, but has been agreed that granular notification updates won't be a thing on the near horizon?

Reply:
I don't think it has been agreed and may well likely happen. I think we should go with the requirements we know though and I don't think it will be too difficult to change to support granular permissions. The `UserNotificationPreferences` resource will still be singular but have collections of create and update permissions or something like that. I think in many ways it will be easier to handle adding further granularity with this approach e.g. if we were to add notifications that a user has been created or something totally unrelated to courses. That may end up a totally different set of tables etc on the backend so would require a co-ordinating class to represent the total preferences.